File: Task-2/Subsequence-String.js
function isHelloSubsequence(S) {
  const target = "hello";
  let index = 0;

  for (let i = 0; i < S.length; i++) {
    if (S[i] === target[index]) {
      index++;
    }

    if (index === target.length) {
      console.log("YES");
      return;
    }
  }

  console.log("NO");
}

// Example usage:
const S1 = "ahhellllloou";
isHelloSubsequence(S1);

const S2 = "hlelo";
isHelloSubsequence(S2);

File: Task-2/Subsequence String.c
#include <stdio.h>
#include <string.h>

#define MAX_SIZE 100000

int main()
{
    char S[MAX_SIZE];
    scanf("%s", S);

    int len = strlen(S);
    char target[] = "hello";
    int targetIndex = 0;

    for (int i = 0; i < len; i++)
    {
        if (S[i] == target[targetIndex])
        {
            targetIndex++;
        }

        if (targetIndex == 5)
        {
            printf("YES\n");
            return 0;
        }
    }

    printf("NO\n");

    return 0;
}
